disttask/framework/dispatcher/dispatcher_manager.go

```diff
@@ -159,11 +156,29 @@ func (dm *Manager) dispatchTaskLoop() {
 			if len(tasks) == 0 {
 				break
 			}
+			waitcnt := make(map[string]int)
+			for _, task := range tasks {
+				waitcnt[task.Type] = 0
+			}
+			for _, task := range tasks {
+				if !dm.isRunningTask(task.ID) {
+					waitcnt[task.Type]++
+					tidbmetrics.DistTaskDispatcherDurationGauge.WithLabelValues(task.Type, tidbmetrics.DpWaitingStatus, fmt.Sprint(task.ID)).Set(float64((time.Now().Sub(task.StartTime)).Microseconds()))
```

Comment:
I think you can use `Inc`/`Dec` instead of `Set`, and deleted the struct `waitcnt`